presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -112,6 +122,246 @@ When a ``GROUP BY`` clause is used in a ``SELECT`` statement all output
 expression must be either aggregate functions or columns present in
 the ``GROUP BY`` clause.
 
+**Complex Grouping Operations**
+
+Presto also supports complex aggregations using the ``GROUPING SETS``, ``CUBE``
+and ``ROLLUP`` syntax. This syntax allows users to perform analysis that require
+aggregation on multiple sets of columns in a single query. Complex grouping
+operations do not support grouping on expressions composed of input columns.
+Only column names or ordinals are allowed.
```

Comment:
How can we use column ordinals? 
When I try to execute `select * from (select 'test' as test) group by grouping sets ((1))` Presto throws syntax error. The grammar file uses _qualifiedName_ reference which can't be a numeric value.

Also, the grouping sets can only be used with input expressions rather than query columns unlike GROUP BY statement. The following query:
`select test as test2 from (select 'test' as test) group by grouping sets ((test2))` throws an exception `Column '"test2"' cannot be resolved`
Is this the intended behaviour or a bug?